PostPairsTrading.md

```diff
@@ -0,0 +1,260 @@
+log priceX = 1.609438 2.302585 1.94591 1.386294 2.079442
+
+log priceY = 3.135494 3.401197 3.218876 3.401197 3.555348
+
+spreads: log priceY - log priceX // = 1.526056 1.098612 1.272966 2.014903 1.475907
+```
+
+---
+
+We're making progress, as we observe **numbers now fluctuating within much smaller ranges**. However, we're still missing a clear understanding of the underlying relationship. While we've normalized the data using logarithms, we now need to align their discrepancies to a single asset. 
+
+Since both assets are related, **we can leverage linear regression** to our advantage. This enables us to simplify our spreads effectively. So, we'll conduct a basic linear regression analysis using historical data to discern the disparity between them:
+
+--- 
+
+$spread = log(priceY) - (beta * log(priceX)+alpha)$
+
+```q
+historical_data_priceX: 7 10 6 5 8
+
+historical_data_priceY: 23 25 16 20 15
+
+beta: 0.2679227
+
+alpha: 2.444817
+```
+
+---
+
+We've already calculated the alpha and beta using the logarithmic values of our historical data (since we don't have prior knowledge of the real-time price values for priceX and priceY). Now, all that remains is to apply the previous formula to derive our spreads:
+
+---
+
+$spreads$ = -0.1493929 0.0451223 -0.08835117 0.0451223 0.1579725
+
+---
+
+And this precisely meets our objective—a **comprehensive method for representing relative changes between both assets**. As we can deduce, our mean is now 0 because our assets are normalized, cointegrated and on the same scale. Therefore, ideally, the differential between their prices should be 0. Consequently, when our spread is below 0, we infer that asset X is overpriced, whereas if it's above 0, then asset Y is overpriced.
+
+![Spreads](https://github.com/hablapps/pairstrading/blob/5-Post/Spreads.gif?raw=true)
+
+
+## Two steps forward, one step back.
+
+Before proceeding to plot the NASDAQ100-SP500 spreads, we need to first plan our algorithm. In this post, we intend to create **a real-time scenario** for Pairs trading, so careful planning is essential for our sake.
+
+Decomposing our steps, let's start from the very beginning:
+
+1. Data management part
+
+        1.1. Read data
+
+        1.2. Filter data
+
+2. Outside the loop
+
+        2.1. Linear regression
+
+        2.2. Initialize values 
+
+3. Inside the loop
+
+        3.1. Calculate spreads
+
+        3.2. Write on buffers
+
+
+As previously mentioned, historical data is crucial for generating accurate spreads. We need to calculate each spread in real-time **using precomputed alpha and beta values** derived from both prices. Therefore, once we have obtained our historical values, we can proceed with calculating our linear regression.
+
+---
+
+```q
+// Fix data and take log(prices) -> Simulated data
+priceX: 0!1_(update delta:0f^deltas dateTime from distinct select distinct dateTime, log bid, log ask from (update dateTime:"P"$@[;19;:;"."] each dateTime from tab1) where not null bid);

```

Comment:
this is a long line, divide this into maybe 3 different lines?

Reply:
and also I think is a good moment to explain briefly qsql. Maybe you can say that q is quite hard to learn, but qsql is a vey good point to start

Reply:
I did explained QSQL but not in this line but before